packages/foam-vscode/src/features/commands/convert-links-format-in-note.ts

```diff
@@ -0,0 +1,179 @@
+import { convertLinkFormat } from '../../core/janitor';
+const vscode = require('vscode'); /* cannot import workspace from above statement and not sure what happened */
+
+enum ConvertOption {
+  Wikilink2MDlink,
+  MDlink2Wikilink,
+}
+
+interface IConfig {
+  from: string;
+  to: string;
+}
+
+const Config: { [key in ConvertOption]: IConfig } = {
+  [ConvertOption.Wikilink2MDlink]: {
+    from: 'wikilink',
+    to: 'link',
+  },
+  [ConvertOption.MDlink2Wikilink]: {
+    from: 'link',
+    to: 'wikilink',
+  },
+};
+
+export default async function activate(
+  context: ExtensionContext,
+  foamPromise: Promise<Foam>
+) {
+  const foam = await foamPromise;
+
+  /* 
+  commands:
+  foam-vscode.convert-wikilink-to-markdownlink-inplace
+  foam-vscode.convert-markdownlink-to-wikilink-inplace
+  foam-vscode.convert-wikilink-to-markdownlink-in-copy
+  foam-vscode.convert-markdownlink-to-wikilink-in-copy
+   */
+  context.subscriptions.push(
+    commands.registerCommand(
+      'foam-vscode.convert-wikilink-to-markdownlink-inplace',
+      () => {
+        return convertLinkInPlace(
+          foam.workspace,
+          foam.services.parser,
+          foam.services.matcher,
+          Config[ConvertOption.Wikilink2MDlink]
+        );
+      }
+    ),
+    commands.registerCommand(
+      'foam-vscode.convert-markdownlink-to-wikilink-inplace',
+      () => {
+        return convertLinkInPlace(
+          foam.workspace,
+          foam.services.parser,
+          foam.services.matcher,
+          Config[ConvertOption.MDlink2Wikilink]
+        );
+      }
+    ),
+    commands.registerCommand(
+      'foam-vscode.convert-wikilink-to-markdownlink-incopy',
+      () => {
+        return convertLinkInCopy(
+          foam.workspace,
+          foam.services.parser,
+          foam.services.matcher,
+          Config[ConvertOption.Wikilink2MDlink]
+        );
+      }
+    ),
+    commands.registerCommand(
+      'foam-vscode.convert-markdownlink-to-wikilink-incopy',
+      () => {
+        return convertLinkInCopy(
+          foam.workspace,
+          foam.services.parser,
+          foam.services.matcher,
+          Config[ConvertOption.MDlink2Wikilink]
+        );
+      }
+    )
+  );
+}
+
+/**
+ * convert links based on its workspace and the note containing it.
+ * Changes happen in-place
+ * @param fWorkspace
+ * @param fParser
+ * @param fMatcher
+ * @param convertOption
+ * @returns void
+ */
+async function convertLinkInPlace(
+  fWorkspace: FoamWorkspace,
+  fParser: ResourceParser,
+  fMatcher: IMatcher,
+  convertOption: IConfig
+) {
+  const editor = window.activeTextEditor;
+  const doc = editor.document;
+
+  if (!isMdEditor(editor) || !fMatcher.isMatch(fromVsCodeUri(doc.uri))) {
+    return;
+  }
+  // const eol = getEditorEOL();
+  let text = doc.getText();
+
+  const resource = fParser.parse(fromVsCodeUri(doc.uri), text);
+
+  const textReplaceArr = resource.links
+    .filter(link => link.type === convertOption.from)
```

Comment:
(note: in my proposed approach, this check would be `link.type !== convertOption.to`)